File: sub_multiple.pl
use strict;


my @patterns;

my ($filename_patterns) = = @ARGV;

open(IN, "<$filename_patterns") or die "$filename_patterns not found.";

while (<IN>) 
{
    chop();
    ($old, $new) = split(\t);
    push $patterns ($old, $new);
}
close(IN);

while (<STDIN>) 
{
    for pattern in $patterns


}

File: sub_multiple.py
import string, re, sys

if __name__ == "__main__":
    param_filename_patterns = sys.argv[1]

    infile = open(param_filename_patterns, "r")

    patterns = []
    for line in infile:
        (old, new) = line[:-1].split("\t")
        patterns.append( (re.compile( "%s" % old), new) )

    infile.close()

    for line in sys.stdin:

        for pattern, new in patterns:
            line = pattern.sub( new, line )

        print line[:-1]
